Makefile

```diff
@@ -169,46 +176,77 @@ external.distclean:
 # Targets
 
 dev: external/build/$(CROSS_TRIPLE)/lib/libcephd.a
+	@$(MAKE) go.init
+	@$(MAKE) go.validate
 	@$(MAKE) go.build
 	@$(MAKE) release.build.containers.$(GOOS)_$(GOARCH)
 
-build: external
+build.common:
+	@$(MAKE) go.init
+	@$(MAKE) go.validate
+	@$(MAKE) external
+
+build: build.common
 	@$(MAKE) go.build
 
-install: external
+install: build.common
 	@$(MAKE) go.install
 
-check test: external
+check test: build.common
 	@$(MAKE) go.test
 
-lint: go.lint
+lint:
+	@$(MAKE) go.init
+	@$(MAKE) go.lint
 
-vet: go.vet
+vet:
+	@$(MAKE) go.init
+	@$(MAKE) go.vet
 
-fmt: go.fmt
+fmt:
+	@$(MAKE) go.init
+	@$(MAKE) go.fmt
 
 vendor: go.vendor
 
 clean: go.clean external.clean
 	@rm -fr $(WORKDIR) $(RELEASE_DIR)/* $(BIN_DIR)/*
 
-container: build
-	@RELEASE_CLIENT_SERVER_PLATFORMS=$(GOOS)_$(GOARCH) build/release/release.sh build containers
-
 distclean: go.distclean clean external.distclean
 
-build.platform.%:
-	@$(MAKE) GOOS=$(word 1, $(subst _, ,$*)) GOARCH=$(word 2, $(subst _, ,$*)) build
+cross.build:
+	@$(MAKE) external
+	@$(MAKE) go.build
+
+cross.build.platform.%:
+	@$(MAKE) GOOS=$(word 1, $(subst _, ,$*)) GOARCH=$(word 2, $(subst _, ,$*)) cross.build
+
+cross.parallel: $(foreach p,$(ALL_PLATFORMS), cross.build.platform.$(p))
 
-cross: $(foreach p,$(ALL_PLATFORMS), build.platform.$(p))
+cross:
+	@$(MAKE) go.init
+	@$(MAKE) go.validate
+	@$(MAKE) cross.parallel
 
 release: cross
 	@$(MAKE) release.build
 
-publish: release
+publish:
+ifneq ($(filter master alpha beta stable, $(CHANNEL)),)
```

Comment:
are there changes to https://github.com/rook/rook/blob/master/build/README.md that should also be part of this PR?